scripts/py_matter_yamltests/matter_yamltests/constraints.py

```diff
@@ -360,6 +360,12 @@ def check_response(self, value, value_type_name) -> bool:
             success = value >= -36028797018963967 and value <= 36028797018963967
         elif self._type == 'nullable_int64s' and type(value) is int:
             success = value >= -9223372036854775807 and value <= 9223372036854775807
+        elif self._type == 'single' and type(value) is float:
```

Comment:
to be super nitpicky, NaN is probably also both a single and a double .... not sure if/how much we care or if that is a real value we should accept/expect (NaN sounds like an optional::missing)